.github/workflows/release.yml

```diff
@@ -0,0 +1,326 @@
+          asset_name: step-cli_${{ needs.create_release.outputs.version }}.tar.gz
+          asset_content_type: application/gzip
+
+  build_upload_darwin_assets:
+    name: Build & Upload Mac OS Darwin Assets
+    runs-on: ubuntu-latest
+    needs: create_release
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Setup Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: '^1.15.6'
+      - name: Build
+        id: build
+        run: |
+          PATH=$PATH:/usr/local/go/bin:/home/admin/go/bin
+          make darwin-artifacts
+      - name: Upload Release Assets
+        id: upload-release-assets
+        run: |
+          tag_name="${GITHUB_REF##*/}"
+          hub release edit $(find ./.releases -type f -printf "-a %p ") -m "" "$tag_name"
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+  build_upload_linux_assets:
+    name: Build & Upload Linux Assets
+    runs-on: ubuntu-latest
+    needs: create_release
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Setup Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: '^1.15.6'
+      - name: APT Install
+        id: aptInstall
+        run: sudo apt-get -y install build-essential libpam-dev debhelper devscripts fakeroot
+      - name: Build
+        id: build
+        run: |
+          PATH=$PATH:/usr/local/go/bin:/home/admin/go/bin
+          make linux-artifacts
+      - name: Upload Release Assets
+        id: upload-release-assets
+        run: |
+          tag_name="${GITHUB_REF##*/}"
+          hub release edit $(find ./.releases -type f -printf "-a %p ") -m "" "$tag_name"
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+  build_upload_windows_assets:
+    name: Build & Upload Windows Assets
+    runs-on: ubuntu-latest
+    needs: create_release
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Setup Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: '^1.15.6'
+      - name: Build
+        id: build
+        run: |
+          PATH=$PATH:/usr/local/go/bin:/home/admin/go/bin
+          make windows-artifacts
+      - name: Upload Release Assets
+        id: upload-release-assets
+        run: |
+          tag_name="${GITHUB_REF##*/}"
+          hub release edit $(find ./.releases -type f -printf "-a %p ") -m "" "$tag_name"
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+  build_upload_docker:
+    name: Build & Upload Docker Images
+    runs-on: ubuntu-latest
+    needs: create_release
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Setup Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: '^1.15.6'
+      - name: Build
+        id: build
+        run: |
+          PATH=$PATH:/usr/local/go/bin:/home/admin/go/bin
+          make docker-artifacts
```

Comment:
Not related to this file, but the build shows this log:
```
29 WARNING! Using -*** the CLI is insecure. Use --password-stdin.
30 docker buildx create --use --name mybuilder --platform="linux/amd64,linux/386,linux/arm,linux/arm64" || true
31 mybuilder
32 WARNING! Your password will be stored unencrypted in /home/runner/.docker/config.json.
33 Login Succeeded
34 Configure a credential helper to remove this warning. See
35 https://docs.docker.com/engine/reference/commandline/login/#credentials-store
```

Reply:
I think we can reduce the amount of time to produce images if we change the way we build those images. If instead of building everything in docker, we just build the binaries in one VM, using GOOS and GOARCH, and then just COPY the appropriate binary to the docker image. We'll get rid of `FROM golang:alpine AS builder`. We are not using CGO so I believe the builds should work properly.

Reply:
Fixed the docker login warning.

Reply:
With regards to speeding up docker image builds that's probably outside the scope of this PR. Let's open up another issue for it.

Reply:
But I like your idea for it.

Reply:
Sounds good.